tests/conftest.py

```diff
@@ -1,71 +1,150 @@
-from dataclasses import dataclass
-from pathlib import Path
-
 import pytest
-
+from typing import Optional
+from pathlib import Path
 from ragger.firmware import Firmware
 from ragger.backend import SpeculosBackend, LedgerCommBackend, LedgerWalletBackend
+from ragger.navigator import NanoNavigator
+from ragger.utils import app_path_from_app_name
 
 
-@dataclass(frozen=True)
-class Application:
-    path: Path
-    firmware: Firmware
-
+# This variable is needed for Speculos only (physical tests need the application to be already installed)
+# Adapt this path to your 'tests/elfs' directory
+APPS_DIRECTORY = (Path(__file__).parent / "elfs").resolve()
 
-ELF_DIRECTORY = Path(__file__).parent / 'elfs'
+# Adapt this name part of the compiled app <name>_<device>.elf in the APPS_DIRECTORY
+APP_NAME = "boilerplate"
 
 BACKENDS = ["speculos", "ledgercomm", "ledgerwallet"]
 
-APPLICATIONS = [
-    Application(ELF_DIRECTORY / 'boilerplate_nanos.elf', Firmware('nanos', '2.1')),
-    Application(ELF_DIRECTORY / 'boilerplate_nanox.elf', Firmware('nanox', '2.0.2')),
-    Application(ELF_DIRECTORY / 'boilerplate_nanosp.elf', Firmware('nanosp', '1.0.3'))
-]
+DEVICES = ["nanos", "nanox", "nanosp", "all"]
+
+FIRMWARES = [Firmware('nanos', '2.1'),
+             Firmware('nanox', '2.0.2'),
+             Firmware('nanosp', '1.0.3')]
 
 
 def pytest_addoption(parser):
-    parser.addoption("--backend", action="store", default="speculos")
+    parser.addoption("--device", choices=DEVICES, required=True)
+    parser.addoption("--backend", choices=BACKENDS, default="speculos")
+    parser.addoption("--display", action="store_true", default=False)
+    parser.addoption("--golden_run", action="store_true", default=False)
+    parser.addoption("--log_apdu_file", action="store", default=None)
 
 
-@pytest.fixture(params=APPLICATIONS)
-def application(request):
-    return request.param
+@pytest.fixture(scope="session")
+def backend_name(pytestconfig):
+    return pytestconfig.getoption("backend")
 
 
 @pytest.fixture(scope="session")
-def backend(pytestconfig):
-    return pytestconfig.getoption("backend")
+def display(pytestconfig):
+    return pytestconfig.getoption("display")
 
 
-def create_backend(backend: str, application: Application):
-    if backend.lower() == "ledgercomm":
-        return LedgerCommBackend(application.firmware, interface="hid")
-    elif backend.lower() == "ledgerwallet":
-        return LedgerWalletBackend(application.firmware)
-    elif backend.lower() == "speculos":
-        assert application.path.is_file(), f"'{application.path}' elf must exist"
-        return SpeculosBackend(application.path, application.firmware)
-    else:
-        raise ValueError(f"Backend '{backend}' is unknown. Valid backends are: {BACKENDS}")
+@pytest.fixture(scope="session")
```

Comment:
Some option doesn't really apply for this tests/conftest.py But I guess you are adding them so that you can latter use the configuration conftest?

Reply:
Yes, in the next PR this conftest is reduced to 10 lines !

After all if a test doesn't want a fixture he doesn't subscribe itself to it and there is no overhead